### PR TITLE
fix: version-range < and <=

### PR DESCRIPTION
**Pull request by mg979**
vim.version.range() couldn't parse them correctly.
For example, vim.version.range('<0.9.0'):has('0.9.0') returned `true`.

### CODE REVIEW

**Review by justinmk — APPROVED** [review]
LGTM except for comment. Thanks!

p.s. @folke maybe worth a glance?

File: runtime/lua/vim/version.lua [hunk 1]
```diff
@@ -226,8 +226,14 @@ function Range:has(version)
   if type(version) == 'string' then
     ---@diagnostic disable-next-line: cast-local-type
     version = M.parse(version)
+  else
+    -- ensures that versions can be compared
+    setmetatable(version, Version)
```
[the remark this answers is not in the document]
Comment (mg979) [reply runtime/lua/vim/version.lua:231]: Storing/restoring the metatable before returning would be cheaper, would be ok?

[the remark this answers is not in the document]
Comment (justinmk) [reply runtime/lua/vim/version.lua:231]: eh... sure :) though I don't think performance should matter here.

File: runtime/lua/vim/version.lua [hunk 2]
```diff
@@ -292,6 +301,12 @@ function M.range(spec) -- Adapted from https://github.com/folke/lazy.nvim
     local to = vim.deepcopy(semver)
     if mods == '' or mods == '=' then
       to.patch = to.patch + 1
+    elseif mods == '<' then
+      from = M._version({})
+      to.patch = to.patch
```
Comment (justinmk) [inline runtime/lua/vim/version.lua:306]: what's the purpose of this assignment?

Comment (mg979) [reply runtime/lua/vim/version.lua:306]: on line 306? an oversight.

**Pull request merged**